pims/base_frames.py

```diff
@@ -528,3 +528,268 @@ def process(img_or_iterable, *args, **kwargs):
+    def sizes(self):
+        """ Returns a dict of all dimension sizes, including x and y. """
+        result = {d.name: d.size for d in self._dims}
+        result['y'], result['x'] = self.frame_shape_2D
+        return result
+
+    @property
+    def aggregate(self):
+        """ This determines which dimensions will be aggregated into one Frame.
+        The ndarray that is returned by get_frame has the same dimension order
+        as the order of aggregate. """
+        return [d.name for d in self._dims if d.aggregate]
+
+    @aggregate.setter
+    def aggregate(self, value):
+        for dim_aggregate in value:
+            if dim_aggregate.lower() in ['x', 'y']:
+                raise ValueError('Aggregate does not take dimensions x or y.')
+            if dim_aggregate not in self.dims:
+                raise ValueError(('Dimension named ''{}'' does not exist ' +
+                                  'in this image.').format(dim_aggregate))
+
+        new_dims_aggr = [0] * len(value)
+        new_dims_rest = []
+        for dim in self._dims:
+            for n_aggr, dim_aggregate in enumerate(value):
+                if dim.name == dim_aggregate:
+                    new_dims_aggr[n_aggr] = dim
+                    dim.aggregate = True
+                    break
+            else:
+                new_dims_rest.append(dim)
+                dim.aggregate = False
+        self._dims = new_dims_rest + new_dims_aggr
+
+    @property
+    def iterate(self):
+        """ This determines which dimensions will be iterated over by the
+        FramesSequence. The last element will iterate fastest. """
+        return [d.name for d in self._dims if d.iterate]
+
+    @iterate.setter
+    def iterate(self, value):
+        for dim_iterate in value:
+            if dim_iterate.lower() in ['x', 'y']:
+                raise ValueError('Iterate does not take dimensions x or y.')
+            if dim_iterate not in self.dims:
+                raise ValueError(("Dimension named '{}' does not exist " +
+                                  "in this image.").format(dim_iterate))
+
+        new_dims_iter = [0] * len(value)
+        new_dims_rest = []
+        for dim in self._dims:
+            for n_iter, dim_iterate in enumerate(value):
+                if dim.name == dim_iterate:
+                    new_dims_iter[n_iter] = dim
+                    dim.iterate = True
+                    break
+            else:
+                new_dims_rest.append(dim)
+                dim.iterate = False
+        self._dims = new_dims_iter + new_dims_rest
+
+    @abstractmethod
+    def get_frame_2D(self, **ind):
+        """ The actual frame reader, defined by the subclassed reader.
+
+        This method should take exactly one keyword argument per dimension,
+        reflecting the index along each dimension. It returns a two dimensional
+        ndarray with shape `frame_shape_2D` and dtype `pixel_type`.
+        """
+        pass
+
+    def get_frame(self, i):
```

Comment:
Don't see exactly what you mean here, but I revised the code strongly so please take a look again at this part